source/base/MessageHandler.cpp

```diff
@@ -289,6 +275,20 @@ void MessageHandler::sendMessage(QString const &toJid, QString const &message, Q
         if (shouldSendMsgStanze == true)
         {
             client_->sendMessage(msg);
+            
+            if(type.compare("txt", Qt::CaseInsensitive) != 0)   // XEP-0066
+            {
+                QString outOfBandElement("");
+                outOfBandElement.append("<x xmlns=\"jabber:x:oob\">");
+                outOfBandElement.append("<url>");
+                outOfBandElement.append(message);
+                outOfBandElement.append("</url>");
+                outOfBandElement.append("</x>");
+
+                std::shared_ptr<Swift::RawXMLPayload> outOfBand =
+                        std::make_shared<Swift::RawXMLPayload>(outOfBandElement.toStdString());
+                msg->addPayload(outOfBand);
+            }        
```

Comment:
Here, you add the oob to the msg object, after it was sent. So, this is useless. 
In my opinion, we should check if the message contains an URL, wait for it to be encrypted (lurchAdapter_->exchangePlainBodyByOmemoStanzas(msg);) and then use the encrypted message within the oob. But I don't know, if the xep 0066 also tries to decrypt the content between the <url> tags. Maybee xep-0066 does not work with omemo encryption at all?

Reply:
The idea was not to use xep-0066 when doing omemo to comply with xep-0454: "_Traditional media sharing with HTTP Upload uses [Out-of-Band Data (XEP-0066)](https://xmpp.org/extensions/xep-0066.html) [[4](https://xmpp.org/extensions/xep-0454.html#nt-idm45559075743152)] to repeat the URL from the body and thereby communicating that the URL is in fact meant as media attchment as opposed a clickable link. For the aesgcm URL scheme no such annotation is necessary as aesgcm URLs are considered unique enough and are never supposed to stand alone in a message."_ 
May I need to rework the code. Any suggestion?

Reply:
Good point. I did not read this.
I would suggest, as the xep also mentioned, to only add the oob stanza if no encryption is used. So, create an else branch of this if 
`            if ( isGroup == false // for now, omemo is only supported on 1o1 messaging
                 && (lurchAdapter_->isOmemoUser(toJid) == true) // the receipient client can handle omemo encryption
                 && (! settings_->getSendPlainText().contains(toJid)) // no force for plain text msg in settings
                 )
`
and put it there in the else. Then the oob stanza only gets added, when sending plain text messages. This should comply with the xep and also remove the privacy issue.

Please make an comment before the oob stanza block and with a content like
// xep-0066. Only add the stanza on plain-text messages, as described int the xep

Reply:
I fixed it